### PR TITLE
fix: [CDS-54545]: modified ui to show imagepath as dropdown for ecr

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>
Screen Shot 2023-03-07 at 5 05 15 PM

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger jest

**Review by satyam-harness — APPROVED** [review]

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePath/ArtifactImagePath.tsx [hunk 1]
```diff
@@ -5,24 +5,111 @@
...
+import type { GitQueryParams, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
+import { EXPRESSION_STRING } from '@pipeline/utils/constants'
+
 import css from '../../ArtifactConnector.module.scss'
 
-function ArtifactImagePath(): React.ReactElement {
+interface ArtifactImagePathProps {
+  connectorRef?: string
+  region?: string
+  artifactType?: string
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePath/ArtifactImagePath.tsx:24]:
```suggestion
  artifactType?: ArtifactTriggerConfig['type']
```

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ECRArtifact/ECRArtifact.tsx [hunk 2]
```diff
@@ -94,7 +94,11 @@ export function ECRArtifact({
                   placeholder={getString('select')}
                 />
               </div>
-              <ArtifactImagePath />
+              <ArtifactImagePath
+                connectorRef={getConnectorIdValue(prevStepData)}
+                region={values?.region}
+                artifactType="ECR"
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ECRArtifact/ECRArtifact.tsx:100]:
```suggestion
                artifactType={ENABLED_ARTIFACT_TYPES.Ecr}
```

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePath/ArtifactImagePath.tsx [hunk 3]
```diff
@@ -5,24 +5,111 @@
...
+        {
+          label: 'No images',
+          value: 'noimages'
+        }
+      ]
+      setImages(imageResponseFormatted)
+    }
+  }, [imagesListData])
+
+  if (artifactType) {
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePath/ArtifactImagePath.tsx:78]:
```suggestion
  if (artifactType === ENABLED_ARTIFACT_TYPES.Ecr) {
```

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePath/ArtifactImagePath.tsx [hunk 4]
```diff
@@ -5,24 +5,111 @@
...
+    connectorRef,
+    region: region,
+    repoIdentifier,
+    branch
+  }
+  const {
+    data: imagesListData,
+    loading: imagesListLoading,
+    refetch: refetchImagesList
+    // error: imagesListError
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePath/ArtifactImagePath.tsx:47]: Do we need not to handle error case?

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]

**Pull request closed without merging**